server/neptune/gateway/event/source/file_fetcher.go

```diff
@@ -0,0 +1,66 @@
+package source
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/google/go-github/v45/github"
+	"github.com/palantir/go-githubapp/githubapp"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"net/http"
+)
+
+type FileFetcher interface {
+	GetModifiedFilesFromCommit(ctx context.Context, repo models.Repo, sha string, installationToken int64) ([]string, error)
+}
+
+type GithubFileFetcher struct {
+	ClientCreator githubapp.ClientCreator
+}
+
+func (g *GithubFileFetcher) GetModifiedFilesFromCommit(ctx context.Context, repo models.Repo, sha string, installationToken int64) ([]string, error) {
+	var files []string
+	nextPage := 0
+	for {
+		opts := github.ListOptions{
+			PerPage: 300,
+		}
+		if nextPage != 0 {
+			opts.Page = nextPage
+		}
+		client, err := g.ClientCreator.NewInstallationClient(installationToken)
+		repositoryCommit, resp, err := client.Repositories.GetCommit(ctx, repo.Owner, repo.Name, sha, &opts)
+		if err != nil {
+			return files, err
+		}
+		if resp.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("error fetching repository contents: %s", resp.Status)
+		}
+		for _, f := range repositoryCommit.Files {
+			files = append(files, f.GetFilename())
+
+			// If the file was renamed, we'll want to run plan in the directory
+			// it was moved from as well.
+			if f.GetStatus() == "renamed" {
+				files = append(files, f.GetPreviousFilename())
+			}
+		}
+		if resp.NextPage == 0 {
+			break
+		}
+		nextPage = resp.NextPage
+	}
+	return files, nil
```

Comment:
This should go in the github package:
https://github.com/lyft/atlantis/tree/release-v0.17.3-lyft.1/server/vcs/provider/github

so it's imported as `github.FileFetcher` and this interface can stay here.